crates/meroctl/src/cli/context/create.rs

```diff
@@ -39,25 +44,45 @@ impl CreateCommand {
 
         let client = Client::new();
 
-        match self {
+        // Manually check for exclusivity between application_id and watch
+        if self.application_id.is_some() && self.watch.is_some() {
+            eyre::bail!("Cannot use --application-id and --watch together");
+        }
+
+        match dbg!(self) {
             CreateCommand {
                 application_id: Some(app_id),
                 watch: None,
                 metadata: _,
             } => {
-                create_context(multiaddr, app_id, &client).await?;
+                create_context(multiaddr, app_id, &client, None).await?;
             }
             CreateCommand {
                 application_id: None,
                 watch: Some(path),
                 metadata,
             } => {
                 let path = path.canonicalize_utf8()?;
+                let application_id =
+                    install_app(multiaddr, path.clone(), &client, Some(metadata.clone())).await?;
+                let context_id = create_context(multiaddr, application_id, &client, None).await?;
 
                 let application_id =
                     install_app(multiaddr, path.clone(), &client, metadata.clone()).await?;
 
-                let context_id = create_context(multiaddr, application_id, &client).await?;
+                watch_app_and_update_context(multiaddr, context_id, path, &client, Some(metadata))
+                    .await?;
+            }
+            CreateCommand {
```

Comment:
I don't like this chaining but didn't find better option.

Reply:
I think we should be able to collapse these and selectively apply defaults, seems we only need to match on context ID

```rust
            CreateCommand {
                application_id: None,
                watch: Some(path),
                context_id,
                metadata,
            } => {
                let path = path.canonicalize_utf8()?;
                let application_id =
                    install_app(multiaddr, path.clone(), &client, metadata.clone()).await?;
                let context_id = match context_id {
                    Some(context_id) => context_id,
                    None => create_context(multiaddr, application_id, &client, Some(context_id)).await?,
                };
                watch_app_and_update_context(multiaddr, context_id, path, &client, metadata)
                    .await?;
            }
```